src/kinect_azure.cc

```diff
@@ -351,7 +357,20 @@ Napi::Value MethodStartListening(const Napi::CallbackInfo& info) {
           jsFrame.colorImageFrame.stride_bytes = k4a_image_get_stride_bytes(color_image);
           jsFrame.colorImageFrame.image_data = new uint8_t[jsFrame.colorImageFrame.image_length];
           uint8_t* image_data = k4a_image_get_buffer(color_image);
-          memcpy(jsFrame.colorImageFrame.image_data, image_data, jsFrame.colorImageFrame.image_length);
+          if (g_customDeviceConfig.flip_BGRA_to_RGBA != true){
+            memcpy(jsFrame.colorImageFrame.image_data, image_data, jsFrame.colorImageFrame.image_length);
+          } else {
+            if (rgba_data == NULL)
+              rgba_data = new uint8_t[jsFrame.colorImageFrame.image_length];
```

Comment:
you're allocating memory here, but it never gets deallocated. When the while loop exits (line 522), you should deallocate this memory by calling delete:

```cpp
if (rgba_data != NULL) {
  delete[] rgba_data;
  rgba_data = NULL;
}
```

Reply:
Thanks for the tip.. I'm a bit of a c++ noob.

I would have through that because rgba_data is only allocated once (when it's current value is NULL) then there wouldn't be a need to deallocate.. as in wouldn't it be more performant to allocate once and reuse than allocate and then deallocate every tick? as I said I'm a c++ noob so I'm interested if this is incorrect logic.

Reply:
Yes, the deallocation would need to happen outside the while loop (so it gets reused inside of the while loop like you are saying). When you allocate memory with the "new" keyword, you have to explicitly deallocate it by using that delete keyword, otherwise you'd be leaking memory.

Appreciate the effort, when that's fixed I can merge it into master :-)

Reply:
Saw you implemented it in your commit - will do a final test tonight and merge :-)